NumLib/Fem/FiniteElement/TemplateIsoparametric.h

```diff
@@ -101,13 +103,40 @@ class TemplateIsoparametric
      */
     template <ShapeMatrixType T_SHAPE_MATRIX_TYPE>
     void computeShapeFunctions(const double* natural_pt, ShapeMatrices& shape,
-                               const unsigned global_dim) const
+                               const unsigned global_dim,
+                               bool is_axially_symmetric) const
     {
         NaturalCoordsMappingType::template computeShapeMatrices<
             T_SHAPE_MATRIX_TYPE>(*_ele, natural_pt, shape, global_dim);
+        computeIntegralMeasure(is_axially_symmetric, shape);
+    }
+
+    double interpolateZerothCoordinate(
+        typename ShapeMatrices::ShapeType const& N) const
+    {
+        auto* nodes = _ele->getNodes();
+        typename ShapeMatrices::ShapeType rs(N.size());
+        for (int i=0; i<rs.size(); ++i) {
+            rs[i] = (*nodes[i])[0];
+        }
+        auto const r = N.dot(rs);
+        return r;
     }
 
 private:
+    void computeIntegralMeasure(bool is_axially_symmetric,
+                                ShapeMatrices& shape) const
+    {
+        if (!is_axially_symmetric) {
+            shape.integralMeasure = 1.0;
+            return;
+        }
+
+        auto const r = interpolateZerothCoordinate(shape.N);
+        shape.integralMeasure =
+            2.0 * 3.14159265358979323846264338327950288419 * r;
```

Comment:
use boost constant of pi like Nori used in his PR, or set it somewhere as a global constant?